hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -453,15 +453,16 @@ private void logAuditEvent(boolean succeeded,
 
   private void appendClientPortToCallerContextIfAbsent() {
     final CallerContext ctx = CallerContext.getCurrent();
-    if (isClientPortInfoAbsent(ctx)) {
-      String origContext = ctx == null ? null : ctx.getContext();
-      byte[] origSignature = ctx == null ? null : ctx.getSignature();
-      CallerContext.setCurrent(
-          new CallerContext.Builder(origContext, contextFieldSeparator)
-              .append(CallerContext.CLIENT_PORT_STR, String.valueOf(Server.getRemotePort()))
-              .setSignature(origSignature)
-              .build());
-    }
+    String origContext = ctx == null ? null : ctx.getContext();
+    byte[] origSignature = ctx == null ? null : ctx.getSignature();
+    //When the request comes from the dfsrouter, origContext must contains CLIENT_PORT_STR. See RouterRpcClient.addClientInfoToCallerContext
+    String clientPort = isClientPortInfoAbsent(ctx) ? CallerContext.CLIENT_PORT_STR :
```

Comment:
I don't find this a problem, Even if someone else sends Client port, we are just logging his actual port in the audit log,

Though I don't like the name of the config mentioning dfsrouter, let it be any client who sends you the client port, if someone sends you a port via client port, we just added the actual client port.

At worst if someone finds serious objections on having this in, We can consider doing this only if the guy sending the request is configured under ```dfs.namenode.ip-proxy-users```